src/lib.rs

```diff
@@ -337,6 +338,12 @@ impl Parser {
                     // Continue collecting bytes into param
                     self.param *= 10;
                     self.param += (byte - b'0') as i64;
+
+                    // Mimic XTerm's overflow-prevention behavior
+                    if self.param > u16::MAX as i64 {
```

Comment:
Rather than limit to u16, could this be u32 or instead just use something like `saturating_add`/`saturating_mul` for self.param?